.ci/CI/src/Bootloader.jl

```diff
@@ -471,49 +537,20 @@ function main()
 
     tools_git_repo = get_git_ci_tools_url_branch()
 
-    if is_cpu
-        add_unit_test_job_yaml!(
-            get(job_yamls, "output-cpu", job_yamls["stdout"]),
-            test_package,
-            unit_test_julia_versions,
-            target_branch,
-            CPU,
-            tools_git_repo,
-            get_unit_test_nightly_baseimage(),
-        )
-    end
-
-    if is_cuda || is_amdgpu
-        for version in ["rc", "nightly"]
-            if version in unit_test_julia_versions
-                @info "Remove unit test version $(version) for GPU tests"
-                filter!(v -> v != version, unit_test_julia_versions)
-            end
+    for (test_type_name, platform) in tests_configurations["unit"]
+        if platform == CPU
+            output_yaml = get(job_yamls, "output-cpu", job_yamls["stdout"])
+        else
+            output_yaml = get(job_yamls, "output-gpu", job_yamls["stdout"])
```

Comment:
It might be a good idea to catch if something completely unexpected goes here? As in, using a specific `else if ...` and then throw in the `else` block.

Reply:
I guaranty, that `job_yamls["stdout"]` exist, otherwise `get()` throws an error independent of if the key asked for exist or not.

Reply:
I meant the check for `if platform == CPU`, i.e. make this `else` to something like `else if platform == AMDGPU || platform == CUDA` and add an extra `else` at the end that throws an `InvalidArgument` or similar.

Reply:
> I meant the check for if platform == CPU, i.e. make this else to something like else if platform == AMDGPU || platform == CUDA and add an extra else at the end that throws an InvalidArgument or similar.

Ah, fear point. I will do it.